Source/Core/InputCommon/ControlReference/FunctionExpression.cpp

```diff
@@ -194,6 +207,292 @@ class WhileExpression : public FunctionExpression
   std::string GetFuncName() const override { return "while"; }
 };
 
+// usage: deadzone(input, amount)
+class DeadzoneExpression : public FunctionExpression
+{
+  virtual bool ValidateArguments(const std::vector<std::unique_ptr<Expression>>& args) override
+  {
+    return 2 == args.size();
+  }
+
+  ControlState GetValue() const override
+  {
+    const ControlState val = GetArg(0).GetValue();
+    const ControlState deadzone = GetArg(1).GetValue();
+    return std::copysign(std::max(0.0, std::abs(val) - deadzone) / (1.0 - deadzone), val);
+  }
+
+  void SetValue(ControlState value) override {}
+  std::string GetFuncName() const override { return "deadzone"; }
+};
+
+// usage: smooth(input, seconds)
+// seconds is seconds to change from 0.0 to 1.0
+class SmoothExpression : public FunctionExpression
+{
+  virtual bool ValidateArguments(const std::vector<std::unique_ptr<Expression>>& args) override
+  {
+    return 2 == args.size();
+  }
+
+  ControlState GetValue() const override
+  {
+    const auto now = Clock::now();
+    const auto elapsed = now - m_last_update;
+    m_last_update = now;
+
+    const ControlState desired_value = GetArg(0).GetValue();
+    const ControlState smooth = GetArg(1).GetValue();
+
+    using FSec = std::chrono::duration<ControlState>;
```

Comment:
This one appears a lot throughout the expressions...any thoughts on declaring it _once_ further up, inside the namespace?

Reply:
Yeah. I'll move the clock typedefs too then.